CMakeLists.txt

```diff
@@ -42,6 +42,15 @@ project(ddX VERSION ${VERSION} LANGUAGES Fortran CXX)
 # Extend path for additional cmake modules
 list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")
 
+# If ddX is a subproject, append ddx_ prefix to all targets.
+if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
+    set(ddx_is_subproject false)
+    set(ddx_ "")
+else()
+    set(ddx_is_subproject true)
+    set(ddx_ "ddx_")
```

Comment:
In the CMake world namespacing is usually done by adding the project name with double colons. For migrating boilerplate CMake between projects I tend to prefer `${PROJECT_NAME}::` over `ddx::`.

Reply:
It is impossible to create targets with :: in their names. So I use prefix with underline.